### PR TITLE
Fix some outdated information/typos in development guide

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [x] Passes `pre-commit run --all-files`

Saw these when going over the development guide

### CODE REVIEW

File: docs/source/develop.rst [hunk 1]
```diff
@@ -52,8 +52,8 @@ non-exhaustive list follows:
 
 Git and GitHub can be challenging at first.  Fortunately good materials exist
 on the internet.  Rather than repeat these materials here, we refer you to
-Pandas' documentation and links on this subject at
-https://pandas.pydata.org/pandas-docs/stable/contributing.html
+pandas' documentation and links on this subject at
+https://pandas.pydata.org/pandas-docs/stable/development/contributing.html
```
Comment (phofl) [inline docs/source/develop.rst:56]: We could also point to the dev-docs. We adjusted things quite regularly recently.

Comment (jrbourbeau) [reply docs/source/develop.rst:56]: I'll defer to you as to which link would be a better source for contributors

**Review by jrbourbeau — COMMENTED** [review]
Thanks @phofl! Happy to merge this as is or we can update the pandas contributing list as you mentioned

File: docs/source/develop.rst (hunk 1 above)
Comment (phofl) [reply docs/source/develop.rst]: The lets point to the dev version, bugs are fixed sooner there

**Review by jrbourbeau — APPROVED** [review]

**Pull request merged**